AsyncTwitch/TwitchConnection.cs

```diff
@@ -14,6 +14,86 @@
 
 namespace AsyncTwitch
 {
+    public class RegisteredPlugin
+    {
+        public event Action<TwitchConnection, TwitchMessage> OnMessageReceived = null;
+        public event Action<TwitchConnection> OnConnected = null;
+        public event Action<TwitchConnection> OnChatJoined = null;
+        public event Action<TwitchConnection, ChatUser> OnChatParted = null;
+        public event Action<TwitchConnection, RoomState> OnRoomStateChanged = null;
+        public event Action<string> OnRawMessageReceived = null;
+
```

Comment:
instead of using beginInvoke every time we iterate over the registered plugins dict, we should ship the whole iteration to a task and make the Invocation Synchronous.